src/m365/teams/commands/cache/cache-remove.ts

```diff
@@ -71,34 +73,59 @@ class TeamsCacheRemoveCommand extends GraphCommand {
         logger.logToStderr('This command will execute the following steps.');
         logger.logToStderr('- Stop the Microsoft Teams client.');
         logger.logToStderr('- Clear the Microsoft Teams cached files.');
-  
+
         const result = await Cli.prompt<{ continue: boolean }>({
           type: 'confirm',
           name: 'continue',
           default: false,
           message: `Are you sure you want to clear your Microsoft Teams cache?`
         });
-        
+
         if (result.continue) {
           await this.clearTeamsCache(logger);
         }
-      }      
-    } 
+      }
+    }
     catch (err: any) {
       this.handleRejectedODataJsonPromise(err);
     }
   }
 
   private async clearTeamsCache(logger: Logger): Promise<void> {
     try {
-      await this.killRunningProcess(logger);
-      await this.removeCacheFiles(logger);
+      const folderExists = await this.checkIfCacheFolderExists(logger);
+      if (folderExists) {
+        await this.killRunningProcess(logger);
+        await this.removeCacheFiles(logger);
+        logger.logToStderr('Teams cache cleared!');
+      }
+      else {
+        logger.logToStderr('Cache folder does not exist. Nothing to remove.');
+      }
     }
     catch (e: any) {
       throw e.message as string;
     }
+  }
+
+  private async checkIfCacheFolderExists(logger: Logger): Promise<boolean> {
+    if (this.verbose) {
+      logger.logToStderr('Checking if cache folder exists');
+    }
+
+    const platform = process.platform;
+    let filePath = '';
+
+    switch (platform) {
+      case 'win32':
+        filePath = `${process.env.APPDATA}\\Microsoft\\Teams`;
+        break;
+      case 'darwin':
+        filePath = `${homedir}/Library/Application Support/Microsoft/Teams`;
+        break;
+    }
```

Comment:
If we ever remove the logic earlier in this command to allow it to run only on win and mac, this might lead to undesired side-effects, because an empty path would point to the current directory (?). It would be safe to add a default clause that throws an exception, so that we won't accidentally delete some other folder on disk.

Reply:
@waldekmastykarz I agree, but aren't we already validating this in the #initValidators? Also, this code will only check if a folder exists. The deletion is done in a different method `removeCacheFiles`, which again specifically mentions the path of the Teams folder on the system running the command. 
What would you suggest? Adding another check when checking if folder exists if we are using either `win32` or `darwin`?

Reply:
Ah, good point. It would be better to have the folder path in one place, so that we won't have the risk that they go out of sync. Ideally, we'd detect it just once based on the platform and once we have it, pass it wherever we need to, so that we don't do the detection in multiple places.

Reply:
I will implement this and give you a shout when it's ready. Thanks for the feedback @waldekmastykarz !

Reply:
Perfect! I appreciate your help @MathijsVerbeeck 👏